### ISSUE TITLE
Blocky / Squarish patterned cloud rendering issue

### ISSUE THREAD

**Issue by SCK57** [Qendolin/better-clouds#57]
Clouds wont render unless I am above them or they are behind blocks

- Mod Version: 1.20.2 1.4.2-1.4.0 1.20.2

- Other Mods: yet-another-config-lib-fabric-3.3.0-beta.1+1.20.2

clouds only appearing behind the blocks as blocky and low res
![bc bug 2](https://github.com/Qendolin/better-clouds/assets/152747822/7811d496-85a1-4704-97c4-c86e0ebca292)

![bc bug 1](https://github.com/Qendolin/better-clouds/assets/152747822/2f2dff51-c761-451a-b2f6-8dfcdd60945e)

only the tops of the clouds render
![bc bug 3](https://github.com/Qendolin/better-clouds/assets/152747822/6041df56-cfd5-4e27-81a1-02e251c4716c)

**Comment by Qendolin** [comment]
That seems like a GPU Driver issue, can u send a screenshot of the F3 Menu?

**Comment by SCK57** [comment]
is this it?
![2023-12-03_14 54 20](https://github.com/Qendolin/better-clouds/assets/152747822/992d917d-0150-4e0f-b6ea-b18f21b84c41)

**Comment by SCK57** [comment]
[debug-trace-2023-12-03_144751.zip](https://github.com/Qendolin/better-clouds/files/13540398/debug-trace-2023-12-03_144751.zip)

**Comment by SCK57** [comment]
yea can you help me do that

**Comment by Qendolin** [comment]
Right, you'll have to install [RenderDoc](https://renderdoc.org/) and please tell me which minecraft launcher you have installed. The following method might not work with every one. It should work with Prism, PolyMC or MultiMC.
Make sure to follow the steps closely, if you have any trouble feel free to ask for help.

Once you have installed RenderDoc:
1. Go to the installation folder and run `qrenderdoc.exe`
2. Figure out where you MC Launcher is installed
3. Under the "Launch Application" tab, click on the three dots "..." on the right of "Executable Path" and select the launcher's `.exe` file.
If you are not sure which file is the `.exe` file you can enable file extensions as explained [here](https://www.howtogeek.com/205086/beginner-how-to-make-windows-show-file-extensions/).
![image](https://github.com/Qendolin/better-clouds/assets/32160662/0a261dff-6a1f-443e-935d-8f4908d2cf17)
4. Under "Capture Options" make sure that "Capture Child Process" is enabled.
![image](https://github.com/Qendolin/better-clouds/assets/32160662/f4cbb5d9-fadf-440e-95d8-32f5c3538375)
5. Press the "Launch" button and your MC Launcher should open. Now start the game.
If everything worked you should see this text in the top-left of the game window:
![image](https://github.com/Qendolin/better-clouds/assets/32160662/4bc8dbb3-f4d5-490d-b03f-c2516fa0be27)
If the text does not appear, the rest will not work.
6. If you go back to RenderDoc, it should look something like this:
![image](https://github.com/Qendolin/better-clouds/assets/32160662/05582571-7460-4adb-9719-2e6473a4de8e)
7. Double-click on the child process, it is probably called `javaw.exe`.
This should open a new tab that looks like this:
![image](https://github.com/Qendolin/better-clouds/assets/32160662/2d68652f-ad9d-41a0-8601-3aa4469319be)
8. Now load a world, look at the sky where clouds should be visible and press "F12" or "Print Screen" on your keyboard
(You can also press the "Capture Frame(s) Immediately" button within renderdoc if F12 doesn't work from some reason)
9. Within RenderDoc a preview of the capture you just took should appear. Right-click it and press "Save".
(Save it to a location where you will find it again)
![image](https://github.com/Qendolin/better-clouds/assets/32160662/eee2ef8a-5bd6-4fbe-a0db-b8e89cd773b7)
10. Upload it to [file.io](https://www.file.io/) or where ever you want and reply with the link so that I can download it.

**Comment by SCK57** [comment]
I'm using the normal Minecraft launcher and for 1. how do I access the installation folder?

**Comment by Qendolin** [comment]
1. Open the Minecraft Launcher
2. Open Task Manager (Ctrl + Shift + Esc)
3. Right Click the entry for the Minecraft Launcher and choose "Open file location"
![image](https://github.com/Qendolin/better-clouds/assets/32160662/5c14dabe-2e51-4e03-8498-db8e10f110f2)

**Comment by Qendolin** [comment]
Hey I've just published a new version: https://modrinth.com/mod/better-clouds/version/1.4.5+1.20.2-R
It simplifies the whole process tremendously.
Just do:
- `/betterclouds:debug renderdoc install`
- `/betterclouds:debug renderdoc load`
- Restart your game
- Look at where clouds **should** be visible
- `/betterclouds:debug renderdoc capture`
- Click the path in chat and upload the capture

**Comment by SCK57** [comment]
github dosn't support the file type so I exported it as a json and zip file but idk if those are useful
[capture.chrome.json](https://github.com/Qendolin/better-clouds/files/13805538/capture.chrome.json)
[zip capture.zip](https://github.com/Qendolin/better-clouds/files/13805542/zip.capture.zip)

**Comment by Qendolin** [comment]
The zip export also generates a `.zip.xml` file which I need to open it. Alternatively you can upload the `.rdc` to https://www.file.io/

**Comment by SCK57** [comment]
ok this should link to the rdc file
https://file.io/1k2CxSvCNVmk

**Comment by Qendolin** [comment]
I am still looking for a fix, in the meantime, can you update your AMD drivers? They are from 2022.

**Comment by Aendar** [comment]
Same issue with me, but I have such visual bug with the [Lighty mod](https://github.com/SchmarrnDevs/Lighty) too.
Here is a RenderDoc [capture file](https://file.io/F3R4IKI90eAV).
I updated my video card drivers about a week ago.

![2024-01-12_15 39 14](https://github.com/Qendolin/better-clouds/assets/96650425/a035a892-2018-4d9a-bd6f-9af4d7b6311d)
![2024-01-12_15 39 16](https://github.com/Qendolin/better-clouds/assets/96650425/a003dacf-5a4a-4f8e-bd37-bdf3b7976a4a)
![2024-01-12_15 39 17](https://github.com/Qendolin/better-clouds/assets/96650425/e081a075-0673-4d98-8500-eda8a1c3651e)

**Comment by Aendar** [comment]
I must apologise, if you only install Better clouds itself and the dependencies then it's fine, I'll try to find out which mod is incompatible now

**Comment by Qendolin** [comment]
@Aendar This does not look like the same error to me, could you please open a new GitHub issue? Also I really need your log file as it contains important information.

**Comment by Aendar** [comment]
Sorry, yeah, I've deleted it bcuz I must create a new issue, give me 10 min and I will upload everything.

**Comment by MartiPuigV** [comment]
I know this is from a while ago, but i have the same issue, with the latest versions of both fabric, fabric api, betterclouds, and YACL. I also updated my driver after reading the comments on this issue, but still no result, im still getting the blurry and pixelated effect on the side of blocks. I however do see the clouds from above.

Any help would be appreciated.

**Comment by Qendolin** [comment]
Can you please upload your game log? If there are any important differences it might help to track down the issue.

**Comment by MartiPuigV** [comment]
Where can i find the game log ? I imagine in the .minecraft, but where ?

**Comment by Qendolin** [comment]
It should be  `.minecraft/logs/latest.log`

**Comment by MartiPuigV** [comment]
[latest.log](https://github.com/Qendolin/better-clouds/files/15176140/latest.log)
Should be this one

**Comment by Qendolin** [comment]
Can you try this custom version for 1.20.4 and see if the issue is still there? Just unzip it to get the .jar file. 
[better-clouds-1.5.8-snapshot+036fa2f3+1.20.4.zip](https://github.com/Qendolin/better-clouds/files/15177039/better-clouds-1.5.8-snapshot%2B036fa2f3%2B1.20.4.zip)

**Comment by MartiPuigV** [comment]
Still the same issue im afraid

**Comment by Qendolin** [comment]
Can you try `/betterclouds:debug fallback stencil_texture true` and `/betterclouds:debug fallback stencil_texture false` and report if either of those fixes the issue?

**Comment by MartiPuigV** [comment]
With the custom mod or the one i used before (found on modrinth) ?

**Comment by MartiPuigV** [comment]
Still the same issue

**Comment by Qendolin** [comment]
Then I'm out of ideas again, sorry

**Comment by MartiPuigV** [comment]
No worries. If you do find a solution, please tell me. I'll try the latest updates of the mod.

**Comment by orangishcat** [comment]
No idea how to fix, but you can try your luck with the new 26.2 Blaze3D renderer.
